### PR TITLE
xGov-6 Transaction Builder UI

### PR DESCRIPTION
**Pull request by No-Cash-7970**
This pull request contains a new proposal to fund the “Transaction Builder UI” (name not final).

The “Transaction Builder UI” (name not final) will be an online tool that provides an easy and user-friendly way to create just about any transaction on Algorand. Creating and sending a transaction on Algorand without an existing tool for that particular type of transaction requires using the SDK or `goal`, which requires programming knowledge and/or a node setup. Just about every wallet makes simple payment transactions easy. However, many of the more advanced transactions are either difficult or impossible within the wallets’ UIs. This creates a large gap where a non-programmer user cannot conveniently use many of Algorand’s more advanced features. The purpose of the Transaction Builder UI is to fill in this gap.

This pull request also adds the `.vscode` directory to the .gitignore file.

### CODE REVIEW

**Comment by No-Cash-7970** [comment]
The amount requested is not a typo. Only 1 Algo of funding is being requested

**Comment by averagezen** [comment]
Could you explain some of the types of transactions that this would solve, and why only 1 ALGO is requested? Seems curious.

**Comment by No-Cash-7970** [comment]
> Could you explain some of the types of transactions that this would solve, and why only 1 ALGO is requested? Seems curious.

I'm hoping to support all types of transactions on Algorand, but I'm especially looking to solve the particular pain points that I've had like key registration (needed for running and maintaining a node) and smart contract opt-in/out.

As for the 1 Algo requested, it's sort of a joke. I wanted to try out the xGov proposal system, and maybe get some feedback along the way. I'm not looking to make money out of this project. I was thinking about doing it for free anyway because I think a transaction builder would be something that would be helpful to me and many others in the Algorand community. It's an opportunity for me to give back to the community.

File: Proposals/xgov-x.md [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+- Show and explain transaction details and the transaction fields, if user desires. The Transaction Builder UI may be used as an education tool.
+- The UI should be easy to use across various devices and screen sizes.
+- Safe defaults. The safest options are set as the default.
+- Allow the user to easily use and switch between MainNet, TestNet, and BetaNet.
+- Allow the user to specify a custom `algod` node.
+
+#### Economics and Finance
+
+- The user must not be charged a fee other than the Algorand network fee(s).
+- The need for a web server should be kept to a minimum. Preferably, the Transaction Builder UI wouldn’t need a back-end web server, so cost of maintaining the Transaction Builder UI is as low as possible.
```
Comment (fifthrace) [inline Proposals/xgov-x.md]: Are you thinking everything will pretty much be client side, so just host then in an s3 bucket or something?

File: Proposals/xgov-x.md [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+
+#### User Interface and User Experience
+
+- Focused. The Transaction Builder UI is solely tool for creating Algorand transactions.
+- Simple. Building an Algorand transaction can get very complex very quickly, so the UI shouldn’t add to that complexity. However, the UI shouldn’t completely hide complex details either. Allow the user to see more of the details if they want to.
+- Show and explain transaction details and the transaction fields, if user desires. The Transaction Builder UI may be used as an education tool.
+- The UI should be easy to use across various devices and screen sizes.
+- Safe defaults. The safest options are set as the default.
+- Allow the user to easily use and switch between MainNet, TestNet, and BetaNet.
+- Allow the user to specify a custom `algod` node.
```
Comment (fifthrace) [inline Proposals/xgov-x.md]: Is the idea that there would be a default already setup for the user to use? Or would you need to bring your own node? Hoping the former.

**Comment by joe-p** [comment]
FYI Shiva is working on this feature in dappflow: https://app.dappflow.org/composer/bundles

**Comment by scholtz** [comment]
Hi @No-Cash-7970 

What about modifing the AWallet so that person can export base64 encoded object of the transaction in dev mode?

Web: https://www.a-wallet.net/
Source: https://github.com/scholtz/wallet

Video on how to make account online from web wallet on one click: https://youtu.be/cSJjmnzvvLY

**Comment by scholtz** [comment]
> FYI Shiva is working on this feature in dappflow: https://app.dappflow.org/composer/bundles


Can you please navigate me where is the current DAppFlow source code pls?

**Comment by joe-p** [comment]
> > FYI Shiva is working on this feature in dappflow: https://app.dappflow.org/composer/bundles
> 
> Can you please navigate me where is the current DAppFlow source code pls?

The latest version's source is not available. The most recent open-source code we have is here: https://github.com/algodesk-io/dappflow

File: Proposals/xgov-x.md (hunk 2 above)
Comment (No-Cash-7970) [reply Proposals/xgov-x.md]: Yes, the default would be like most DApps where a service like AlgoNode is used. Advanced users can choose to use a different node or their own node.

**Comment by No-Cash-7970** [comment]
@scholtz 
> Hi @No-Cash-7970
> 
> What about modifing the AWallet so that person can export base64 encoded object of the transaction in dev mode?
> 
> Web: https://www.a-wallet.net/ Source: https://github.com/scholtz/wallet
> 
> Video on how to make account online from web wallet on one click: https://youtu.be/cSJjmnzvvLY

I'm not familiar with AWallet, but I'll look into it. Thanks for the the suggestion!

File: Proposals/xgov-x.md (hunk 1 above)
Comment (urtho) [reply Proposals/xgov-x.md]: How about pushing the page to IPFS and minting an ARC-19 NFT that points to the latest version ?
The content will be cached by gateways dedicated to Algorand NFTs so no need to pin it. 

Also this transaction template (ARC19) could be part of the page so others could host such SPAs the WEB3 way as well.

File: Proposals/xgov-6.md [hunk 3]
```diff
@@ -0,0 +1,108 @@
+---
+period: 8
+title: Transaction Builder UI
+author: No-Cash-7970 (@No-Cash-7970)
+company_name: <None>
+category: Tools
+focus_area: User Onboarding
+open_source: Yes
+amount_requested: 1
+status: Draft
```
Comment (SudoWeezy) [inline Proposals/xgov-6.md:10]: Remember that you need to change this to "Final" if you want your proposal to go through the next voting session.

File: Proposals/xgov-6.md
Comment (Premarko) [inline Proposals/xgov-6.md:1]: Confirming proposal

**Comment by nonfungibleab** [comment]
Congratulations! xGov voting session 1 is now closed and your proposal has passed. Thanks for participating in the process. Should we discuss whether you want to sign a contract to receive 1 A, it's a cumbersome process for such a small amount.

**Comment by Premarko** [comment]
I don't understand this email.

Regards
Marko Prezelj

**Comment by nonfungibleab** [comment]
The message above is for No-Cash-7970

File: Proposals/xgov-6.md
Comment (mightyp15) [reply Proposals/xgov-6.md:1]: Yes

**Pull request merged**